documentation/Installing-Clear-Containers-on-Centos-7.md

```diff
@@ -0,0 +1,203 @@
+# Install the Clear Containers runtime on Centos 7
+
+
+## Introduction
+
+Installing Clear Containers on CentOS is slightly more difficult than on other distributions mainly due to the fact that it does not provide current versions of gcc and glibc.
+
+Hence, this guide explains how to install updated packages in `/usr/local`, to avoid conflicting with the base packages.
+
+## Warning
+
+Note carefully the comment in the section [Install container kernel and rootfs](#install-container-kernel-and-rootfs). You MUST update the packages listed regularly to ensure your Clear Containers benefit from performance improvements, new features and most importantly CVE security fixes.
+
+## Install updated glib and json-glib
+
+Note: these packages need to be built with the default CentOS compiler since they don't build by default with gcc 6.
+
+```
+$ export PKG_CONFIG_PATH=/usr/local/lib/pkgconfig:$PKG_CONFIG_PATH
+$ sudo yum groupinstall 'Development Tools'
+$ sudo yum install zlib-devel libffi-devel
+
+$ curl -L -O "https://download.gnome.org/sources/glib/2.46/glib-2.46.2.tar.xz"
+$ tar -xvf glib-2.46.2.tar.xz
+$ pushd glib-2.46.2
+$ ./configure --prefix=/usr/local --disable-silent-rules && make -j5 && sudo make install
```

Comment:
Just a side note - `-j5` is somewhat system dependent. Not a blocker, and I think we'll leave them in - but more generic would be no `-j` at all or just `-j` with no argument - both of which also have their downsides.